Mage/src/main/java/mage/abilities/common/PayMoreToCastAsThoughtItHadFlashAbility.java

```diff
@@ -14,14 +16,22 @@
  */
 public class PayMoreToCastAsThoughtItHadFlashAbility extends SpellAbility {
 
-    private final ManaCosts costsToAdd;
+    private final Cost costsToAdd;
 
     public PayMoreToCastAsThoughtItHadFlashAbility(Card card, ManaCosts<ManaCost> costsToAdd) {
-        super(card.getSpellAbility().getManaCosts().copy(), card.getName() + " as though it had flash", Zone.HAND, SpellAbilityType.BASE_ALTERNATE);
+        this(card, (Cost) costsToAdd);
+        CardUtil.increaseCost(this, costsToAdd);
+    }
+
+    public PayMoreToCastAsThoughtItHadFlashAbility(Card card, Costs<Cost> costsToAdd) {
+        this(card, (Cost) costsToAdd);
+        this.addCost(costsToAdd);
+    }
+
+    public PayMoreToCastAsThoughtItHadFlashAbility(Card card, Cost costsToAdd) {
+        super(card.getSpellAbility().getManaCosts().copy(), card.getName(), Zone.HAND, SpellAbilityType.BASE_ALTERNATE);
```

Comment:
why did you remove the "as though it had flash" from the second argument to the super constructor?

Reply:
There was a small issue. When "as thought it had flash" string is used both in super constructor and in `getRule()` method, you can see tautology in spell ability choice dialog like:
"You may cast Tegwyll's Scouring as thought it had flash: You may cast Tegwyll's Scouring _as thought it had flash as though it had flash_ if you pay <...>"
So I can't remove this words from the `getRule()` and it was the easiest way to fix. But now it needs to scroll through a long line of text to recognize ability as flash... So I'm doubtful about it.